Sample.java

```diff
@@ -1,7 +1,106 @@
-// Time Complexity :
-// Space Complexity :
+// Problem: Find Largest Value in Each Tree Row
+// Time Complexity : O(n)
+// Space Complexity : O(max width of the tree)
 // Did this code successfully run on Leetcode :
 // Any problem you faced while coding this :
 
 
-// Your code here along with comments explaining your approach
\ No newline at end of file
+// Recursive DFS appraoch
+class Solution {
+    public List<Integer> largestValues(TreeNode root) {
+        
+    
+        
+        List<Integer>  ls = new ArrayList<>();
+        if(root == null) return ls;
+        
+        helper(root, 0, ls);
+        return ls;
+    }
+    
+    private List<Integer> helper(TreeNode root , int level, List<Integer> ls){
+        //base
+        if(root == null) return null;
+        if(level == ls.size()){
+            ls.add(root.val);
+        }else{
+            ls.set( level, Math.max(ls.get(level), root.val));
+        }
+        
+       helper(root.left, level+1, ls);
+       helper(root.right, level+1, ls);
+       return ls;
+        
+    }
+    
+}
+// Problem: Find Largest Value in Each Tree Row
+// Time Complexity : O(n)
+// Space Complexity : O(max width of the tree)
+// Did this code successfully run on Leetcode :
+// Any problem you faced while coding this :
+
+class Solution {
+    public List<Integer> largestValues(TreeNode root) {
+        
+        List<Integer> ans = new ArrayList<>();
+        //edge
+        if(root == null) return ans;
+        
+        Queue<TreeNode> q = new LinkedList<>();
+        int max = Integer.MIN_VALUE;
+        q.add(root); // [5,3,9]
+        
+        while(!q.isEmpty()){
+            
+            int size = q.size();  //3       
+            for(int i=0; i< size; i++){
+                TreeNode cur = q.poll(); //[5]
+                
+                max = Math.max(cur.val, max); // 3
+                
+                if(cur.left != null) q.add(cur.left);
+                if(cur.right != null) q.add(cur.right);
+            }
+            
+            ans.add(max);
+            max = Integer.MIN_VALUE; // resetting the max value
+        }
+        
+        return ans;
+    }
+}
+
+// Problem:2: Find Largest Value in Each Tree Row
+// Time Complexity : O(m*n)

```

Comment:
O(1)

Reply:
I am not clear how the time complexity id O(1) here?

Reply:
board is of finite dimensions 9 by 9. m and n values

Reply:
Got it.